adapters/repos/db/shard_write_batch_objects.go

```diff
@@ -329,46 +321,33 @@ func (ob *objectsBatcher) storeAdditionalStorageWithAsyncQueue(ctx context.Conte
 			continue
 		}
 
-		if hasTargetVectors {
-			for targetVector, vector := range object.Vectors {
-				targetVectors[targetVector] = append(targetVectors[targetVector], &common.Vector[[]float32]{
-					ID:     status.docID,
-					Vector: vector,
-				})
-			}
-			for targetVector, vector := range object.MultiVectors {
-				targetVectors[targetVector] = append(targetVectors[targetVector], &common.Vector[[][]float32]{
-					ID:     status.docID,
-					Vector: vector,
-				})
-			}
-		} else {
-			if len(object.Vector) > 0 {
-				vectors = append(vectors, &common.Vector[[]float32]{
-					ID:     status.docID,
-					Vector: object.Vector,
-				})
-			}
+		for targetVector, vector := range object.Vectors {
+			targetVectors[targetVector] = append(targetVectors[targetVector], &common.Vector[[]float32]{
+				ID:     status.docID,
+				Vector: vector,
+			})
 		}
+		for targetVector, vector := range object.MultiVectors {
+			targetVectors[targetVector] = append(targetVectors[targetVector], &common.Vector[[][]float32]{
+				ID:     status.docID,
+				Vector: vector,
+			})
+		}
+
+		// use empty string for legacy vector, downstream code will handle that appropriately
+		targetVectors[""] = append(targetVectors[""], &common.Vector[[][]float32]{})
```

Comment:
hmm why are you skipping vector from `object.Vector`?

Reply:
good catch, fixed this bug